File: frontend/src/services/health.ts
import { useQuery } from "@tanstack/react-query";
import { apiFetch } from "../lib/api";

export interface HealthStatus {
  status: "healthy" | "degraded";
  database: "healthy" | "unhealthy";
  ai: "configured" | "demo_mode";
  payment_provider: "razorpay_test" | "mock";
  version: string;
}

export function useHealth() {
  return useQuery({
    queryKey: ["health"],
    queryFn: () => apiFetch<HealthStatus>("/health"),
    refetchInterval: 30_000,
  });
}

File: backend/app/api/health.py
from fastapi import APIRouter, Depends
from sqlalchemy import text
from sqlalchemy.orm import Session

from app.core.config import Settings, get_settings
from app.db.session import get_db

router = APIRouter(tags=["health"])


@router.get("/health")
def health(db: Session = Depends(get_db), settings: Settings = Depends(get_settings)):
    db_status = "healthy"
    try:
        db.execute(text("SELECT 1"))
    except Exception:
        db_status = "unhealthy"

    return {
        "status": "healthy" if db_status == "healthy" else "degraded",
        "database": db_status,
        "ai": "configured" if settings.AI_PROVIDER != "mock" else "demo_mode",
        "payment_provider": settings.payment_mode_label,
        "version": settings.VERSION,
    }
